umake/ui/cli/__init__.py

```diff
@@ -157,6 +157,60 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def list_frameworks(listing, detailled=True):
+    if listing == "all" or listing == "available":
```

Comment:
listing is thus not a parameter anymore as we always return everything from this function. However, if you need to use something similar in the future, you can have a look at https://github.com/ubuntu/ubuntu-make/blob/master/umake/tools.py#L51 to see how we define a limited set of choice for a parameter.